arrow/arrow.py

```diff
@@ -76,41 +68,43 @@ class Arrow(object):
 
     def __init__(
         self,
-        year,
-        month,
-        day,
-        hour=0,
-        minute=0,
-        second=0,
-        microsecond=0,
-        tzinfo=None,
-        **kwargs
-    ):
+        year: int,
+        month: int,
+        day: int,
+        hour: int = 0,
+        minute: int = 0,
+        second: int = 0,
+        microsecond: int = 0,
+        tzinfo: Optional[Union["dt_tzinfo", tzfile]] = None,
+        **kwargs,
+    ) -> None:
         if tzinfo is None:
             tzinfo = dateutil_tz.tzutc()
         # detect that tzinfo is a pytz object (issue #626)
         elif (
             isinstance(tzinfo, dt_tzinfo)
             and hasattr(tzinfo, "localize")
             and hasattr(tzinfo, "zone")
-            and tzinfo.zone
+            and tzinfo.zone  # type: ignore
         ):
-            tzinfo = parser.TzinfoParser.parse(tzinfo.zone)
-        elif util.isstr(tzinfo):
-            tzinfo = parser.TzinfoParser.parse(tzinfo)
+            tzinfo = parser.TzinfoParser.parse(tzinfo.zone)  # type: ignore
+        elif isinstance(tzinfo, str):
+            tzinfo = parser.TzinfoParser.parse(tzinfo)  # type: ignore
 
         fold = kwargs.get("fold", 0)
 
         # use enfold here to cover direct arrow.Arrow init on 2.7/3.5
-        self._datetime = dateutil_tz.enfold(
+        self._datetime = dateutil_tz.enfold(  # type: ignore
```

Comment:
Any reason for the ignore here?

Reply:
Yes, indeed, mypy keeps complaining about the following:

```
arrow/arrow.py:97: error: Module has no attribute "enfold"
```

I believe it's the way things are imported in dateutil that throws mypy out of whack.

Reply:
dateutil's imports are slightly confusing, they give pycharm issues as well.

Reply:
Ah yeah for sure, I encountered dateutils annoying imports when using rrule.